packages/liveblocks-client/src/AuthToken.ts

```diff
@@ -93,6 +93,7 @@ export function isRoomAuthToken(data: JsonObject): data is RoomAuthToken {
   return (
     typeof data.appId === "string" &&
     typeof data.roomId === "string" &&
+    // TODO: throw specific error if actor is not a number
```

Comment:
FYI, this function is a [TypeScript predicate function](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates) and as such it should never throw. Other projects (e.g. liveblocks-cloudflare) also rely on this implementation. That said, I agree that throwing targeted error messages is a Good Idea™, just saying that the throwing cannot happen inside a predicate function.

Reply:
I've actually not thrown an error but rather made the error message easier to understand https://github.com/liveblocks/liveblocks/pull/347/files#diff-2f8a3dfc5fdb3180961ecbac8995ec77c35765b7bdbd9957e4c9af62f620bb01R140

From what I understand, if we reach this point, the only way for it to fail is that the token is from a custom callback which was misused (room was not provided)

Reply:
Nice 👍 ! You can remove this TODO comment here.